### PR TITLE
fix(GPO): make default gas tip configurable

### PR DESCRIPTION
**Pull request by colinlyguo**
## 1. Purpose or design rationale of this PR

This PR makes the default gas tip configurable. The default value has been updated to 100 wei (from previous 2 wei) for the following reasons:
- Prevents issues with very low tip values (e.g., 1 wei):
    - Transaction pools often reject transactions with the same tip cap.
    - Extremely low tips (e.g., 1 wei) can fail to increase due to rounding issues in integer arithmetic (e.g., 1 wei * 1.5 = 1 wei).
- Negligible cost:
    - A gas tip of 100 wei is insignificant compared to the base fee under normal conditions.

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] fix: A bug fix

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [x] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [x] This PR is not a breaking change

## Summary by CodeRabbit

- **New Features**
	- Updated minimum tip cap for gas price suggestions from 1 wei to 100 wei to enhance transaction processing.
	- Introduced a new command-line flag to specify a default minimum gas tip cap for transactions.

- **Bug Fixes**
	- Adjusted expected gas price behavior in tests to reflect the updated minimum tip cap.

- **Chores**
	- Incremented patch version from 28 to 29 for the current release.

### CODE REVIEW

File: eth/gasprice/gasprice.go [hunk 1]
```diff
@@ -195,13 +195,14 @@ func (oracle *Oracle) SuggestTipCap(ctx context.Context) (*big.Int, error) {
...
-		//    In the future we can set the price to 1 wei.
-		price := big.NewInt(2)
+		// Before Curie (EIP-1559), we need to return the total suggested gas price (base fee + tip cap).
+		// After Curie, a minimum tip cap of 100 wei is used to avoid issues with extremely low values like 1 wei:
+		// 1. txpool protections:
+		//   - txpool commonly prevents replacing transactions with the same tip cap.
+		//   - A 1 wei tip may fail to increase (e.g., 1.1x rounds down to 1 wei) due to integer arithmetic.
+		// 2. Negligible cost:
+		//   - Both 100 wei and 1 wei are almost zero in cost, so this change has no noticeable impact on users.
+		price := big.NewInt(100)
```
Comment (Thegaram) [inline eth/gasprice/gasprice.go]: Should we make this configurable?

**Review by Thegaram — DISMISSED** [review]

File: eth/gasprice/gasprice.go (hunk 1 above)
Comment (colinlyguo) [reply eth/gasprice/gasprice.go]: sounds good.

Comment (colinlyguo) [reply eth/gasprice/gasprice.go]: supported in this commit: [d1a249e](https://github.com/scroll-tech/go-ethereum/pull/1096/commits/d1a249e916bdfe3a32c8a9dfe824a7175f45bd99).

File: eth/gasprice/gasprice.go [hunk 2]
```diff
@@ -133,6 +141,11 @@ func NewOracle(backend OracleBackend, params Config) *Oracle {
 		defaultBasePrice = DefaultBasePrice
 		log.Warn("Sanitizing invalid gasprice oracle default base price", "provided", params.DefaultBasePrice, "updated", defaultBasePrice)
 	}
+	defaultGasTipCap := params.DefaultGasTipCap
+	if defaultGasTipCap == nil || defaultGasTipCap.Int64() < 0 {
```
Comment (jonastheis) [inline eth/gasprice/gasprice.go]: Just to make sure: we allow setting it to 0?

Comment (colinlyguo) [reply eth/gasprice/gasprice.go]: similar to `DefaultBasePrice` (0 wei), it's valid for a txn with 0 gas tip, too. There are some gotchas to block these txns in tx pool's default setting (e.g. `DefaultTxPoolConfig.PriceLimit` for remote txns).

so for engineering tricks, it's also a good practice to make it non-zero. I'm OK with both. which do you prefer?

Comment (colinlyguo) [reply eth/gasprice/gasprice.go]: after rethinking, it's more practical to enforce positive gas tip, thus changed by this commit: [5e3cbb3](https://github.com/scroll-tech/go-ethereum/pull/1096/commits/5e3cbb30f8e1ae9885b5df8be63807b2d85cefdb)

**Review by Thegaram — APPROVED** [review]

**Review by jonastheis — APPROVED** [review]

**Pull request merged**